iterreg/sparse/solvers.py

```diff
@@ -220,13 +229,16 @@ def ista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=1,
 
 # a priori jitting does not improve anything here
 # @njit
-def fista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=1,
-          verbose=False):
+def fista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=10,
+          w_init=None, verbose=False):
     """Accelerated proximal gradient descent for the Tikhonov problem."""
     p = X.shape[1]
     L = norm(X, ord=2) ** 2
-    w = np.zeros(p)
-    z = np.zeros(p)
+    if w_init is None:
+        w = np.zeros(p)
+    else:
+        w = w_init.copy()
+    z = w.copy()
```

Comment:
I think this opens a new (interesting !) question and should be implemented in another PR. if you revert this for me we can merge